tutorials/cleaning-up-at-scale/index.md

```diff
@@ -0,0 +1,183 @@
+
+This solution leverages the following GCP components: 
+
+*   **Compute Engine**
+*   **Cloud Scheduler**
+*   **Pub/Sub** 
+*   **Cloud Functions**
+
+![High-level overview of the solution](/images/overview.svg "High-level overview of the solution")
+
+**Figure 1: **High-level overview of the solution**
+
+### **How does it work**? 
+
+Each GCE instance in scope is assigned two labels: 
+
+*   **TTL** (Time to Live): which indicates (in minutes) after how much time this VM will not be needed and can be deleted.
+*   **ENV** : it indicates that the instance is part of the pool of VMs that can be checked regularly and can be deleted if TTL is reached. 
+
+The overall flow is the following: 
+
+
+1. We create a Cloud Scheduler cron job which will be triggered regularly (e.g each 5 minutes). The config of the Cloud scheduler contains the values of the label of the pool of VMs to target (e.g ENV= test). Optionally, it can also contain the name of the zone where the VMs to target are deployed. The format is the following: 
+
+		'{"zone":"us-west1-b", "label":"env=test"}'
+
+
+
+2. Once the cron job is triggered, Cloud Scheduler will push a message with the payload above to a Pub/Sub topic. 
+3. A Cloud Function is subscribed to the Pub/Sub topic. Each time triggered, it will perform the following: 
+    1.  Read the payload of the Pub/Sub message
+    2. Extract the **label** and the **zone**
+    3. Filter all the GCE instances which have the label in question ( the zone could also be used to as a filter)
+    4. Iterate through instance and perform the following: 
+        1. Read the value of label **TTL** of each instance 
+        2. Calculate **Delta** = the difference between the current time and the creation time of each instance 
+        3. If **Delta> TTL** , the instance will be deleted. If not, nothing will be done.
+
+
+### Prerequisites
+
+* If you don’t already have one, create a
+    [Google Account](https://accounts.google.com/SignUp).
+
+* Create a Developers Console project.
+    1. In the [Google Developers Console](https://console.developers.google.com/project), select
+      **Create Project**.
+    2. [Enable the Pub/Sub API](https://console.cloud.google.com/flows/enableapi?apiid=pubsub&redirect=https://console.cloud.google.com)
+    3. [Enable the App Engine Admin API](https://console.cloud.google.com/flows/enableapi?apiid=appengine&redirect=https://console.cloud.google.com).  This is required by Cloud Scheduler.
+    4. Visit the [Compute Engine instances](https://console.cloud.google.com/compute/instances) page, this will activate the API.
+    5. [Enable Project Billing](https://support.google.com/cloud/answer/6293499#enable-billing)
+    6. Using [Cloud Shell][https://cloud.google.com/shell/docs/using-cloud-shell], create an App Engine app. This is required by Cloud Scheduler:
+
+           $ gcloud app create --region=us-central
```

Comment:
you mean "us-central1"

Reply:
For historical reasons GAE uses `us-central` without the 1, see `gcloud app regions list`.

Reply:
I see, thanks